gateway/ln-gateway/src/lib.rs

```diff
@@ -198,6 +198,12 @@ pub struct LightningContext {
 // A marker struct, to distinguish lock over `Gateway::clients`.
 struct ClientsJoinLock;
 
+enum HandleStreamResult {
```

Comment:
Let's add a doc comment on this enum.

Also, I think we need a different name, I don't think its descriptive enough. `Result` is a bit weird here too, considering its not a normal `Result`. Perhaps `ReceivePaymentStreamAction`?

Reply:
Done.